### DISCUSSION TITLE
no hay similitud con arduino

### DISCUSSION THREAD

**Discussion by callecuatro1976** [PinguinoIDE/pinguinoide.github.io#38] [Pinguino IDE]
int blue =10;
int green =12;
int red =17;
int i=0;

struct RGB 
{
byte r;
byte g;
byte b;
};

 struct RGB rgb[] = {{  0,  0,  0},  // 0   la nada
                        {255,  0,  0},  // 1   rojo
                        {  0,255,  0},  // 2   verde
                        {  0,  0,255},  // 3   azul
                        {  0,240,255},  // 4   cyan
                        {255,  0,125},  // 5   rosa
                        {100,  0,255},  // 6   morado
                        {255,220,  0},  // 7   amarillo
                        {255, 80,  0},  // 8   naranja
                        {255,255,255},  // 9   blanco
                        {255,255,255}}; // 10  blanco
                        

                        
  void setup()
  {
  pinMode(red, OUTPUT);
    pinMode(blue, OUTPUT);
      pinMode(green, OUTPUT);
  
  }
  void loop()
  {
  
  for (  i=0; i<11; i++) {
  
  setColor(rgb[i]);
  delay(1000);
  
  }
  
  
  
  }

     void setColor(RGB color )
  {
  
  analogWrite(red, color.r);
  analogWrite(green, color.g);
  analogWrite(blue, color.b);
  
  
  }``

en arduino me funciona bien , en pinguino me da error , por que

**Comment by callecuatro1976** [comment]
int tipo_color;
int pin_r =  11;
int pin_g =  12;
int pin_b =  10;
int contador;
int i=0;

struct RGB {int pin_r, pin_g, pin_b;};

const struct RGB rgb[]={
                        {  0,   0,   0},  // 0   la nada
                        {255,   0,   0},  // 1   rojo
                        {  0, 255,   0},  // 2   verde
                        {  0,   0, 255},  // 3   azul
                        {  0, 240, 255},  // 4   cyan
                        {255,   0, 125},  // 5   rosa
                        {100,   0, 255},  // 6   morado
                        {255, 220,   0},  // 7   amarillo
                        {255,  80,   0},  // 8   naranja
                        {255, 255, 255},  // 9   blanco
                        {255, 255, 255}
                       }; // 10  blanco
                       
struct RGB pwm = {0,0,0} ;

INT_TMR2
void  color()
{
int contador=0x00;


    
    if(contador==0x00)      PORTC (0xFF)    ;
    else {
       if(contador==pwm.r)  analogWrite ( pin_r, r ) ;

        if(contador==pwm.g)   analogWrite ( pin_g, g) ;

        if(contador==pwm.b)  analogWrite ( pin_b, b) ;
    }
    contador-- ;
}



void setup ()  {
  pinMode ( pin_r, INPUT ) ;
  pinMode ( pin_g, INPUT ) ;
  pinMode ( pin_g, INPUT ) ;
}

void loop ()  {
for ( i=0; i<11; i++) {
  color(rgb[i].pin_r, rgb[i].pin_g, rgb[i].pin_b);
  delay(1000);
}

}``

**Comment by luzemario** [comment]
¿Que error?
